tsconfig.json

```diff
@@ -2,6 +2,9 @@
   "compileOnSave": false,
   "compilerOptions": {
       "declaration": true,
+      "lib": [
+        "esnext"
+      ],
```

Comment:
While running the PR branch, it seems as though this pull request requires upgrades to Node environment, with the usage of functions like `flatMap` requiring Node 11+ unless this line is removed from tsconfig

Reply:
@camsjams if it were me, I'd forego the use of `flatMap()` altogether.  A single `reduce()` (as opposed to `flatMap().map()`) would solve the problem, I think, and would be better supported.

Reply:
Yup, I upgraded it because of `flatMap` at the time.